src/bundler/bundle_v2.zig

```diff
@@ -4862,6 +4862,8 @@ pub const LinkerContext = struct {
                 // does it have a JS AST?
                 if (!(id < import_records_list.len)) continue;
 
+                _ = this.validateTLA(id);
```

Comment:
Since most projects do not use any top-level await, maybe we can add a boolean in ParseGraph or LinkerGraph that tracks whether any modules at all used top-level await, and then only run this nested for loop over all the import records of all the files when any file has top-level await? We can set that boolean as we receive results from the parser threads to avoid any extra passes